tests/testing_common.py

```diff
@@ -1164,7 +1164,10 @@ def _test_training_gradient_checkpointing(self, model_id, config_cls, config_kwa
 
         for n, param in model.named_parameters():
             if "prompt_encoder." in n:  # prompt tuning methods
-                assert param.grad is not None
+                if not issubclass(config_cls, CPTConfig):
```

Comment:
As a reader might not know that `delta_embedding` is associated with CPT, could you please add a comment to explain the presence/absence of grads here?

Reply:
We have two embeddings in the model. delta_embedding is the only one that should be updated with grads.
I added a comment in the code.